rpc_jobs/standard_job_release.yml

```diff
@@ -78,10 +78,203 @@
+                  value: ghprbGhRepository,
+                ],
+                [
+                  $class: "StringParameterValue",
+                  name: "pr_number",
+                  value: ghprbPullId,
+                ],
+              ]
+            )
+            build(
+              job: "Merge-Pull-Request",
+              wait: false,
+              parameters: [
+                [
+                  $class: "StringParameterValue",
+                  name: "RPC_GATING_BRANCH",
+                  value: RPC_GATING_BRANCH,
+                ],
+                [
+                  $class: "StringParameterValue",
+                  name: "pr_repo",
+                  value: ghprbGhRepository,
+                ],
+                [
+                  $class: "StringParameterValue",
+                  name: "pr_number",
+                  value: ghprbPullId,
+                ],
+              ]
+            )
           } // if
         } // dir
       } // globalWraps
 
+- job:
+    name: 'Component-Release'
+    project-type: pipeline
+    concurrent: false
+    properties:
+      - build-discarder:
+          days-to-keep: 30
+    parameters:
+      - rpc_gating_params
+      - text:
+          name: component_text
+      - validating-string:
+          name: pr_repo
+          regex: "[a-zA-Z0-9-]+/[a-zA-Z0-9._-]+"
+          msg: "The release repository must be supplied."
+      - validating-string:
+          name: pr_number
+          regex: "[0-9]+"
+          msg: "The number of the pull request to merge is required."
+    dsl: |
+      library "rpc-gating@${RPC_GATING_BRANCH}"
+      common.globalWraps(){
+        dir("${env.WORKSPACE}/rpc-metadata"){
+          common.clone_with_pr_refs(
+            ".", "git@github.com:${pr_repo}.git", "origin/pr/${pr_number}/merge",
+          )
+          sh """\
+            #!/bin/bash -xe
+            virtualenv --python python3 .venv3
+            set +x; . .venv3/bin/activate; set -x
+            pip install -c constraints.txt -r requirements.txt
+            pushd rpc_component
+                pip install .
+            popd
+          """
+          component = readYaml text: component_text
+          println "=== component CLI standard out ==="
+          println component_text
+
+          String pred_component_text = sh(
+            script: """\
+              #!/bin/bash -xe
+              set +x; . .venv3/bin/activate; set -x
+              src_branch="origin/master"
+              pr_branch="HEAD"
+              component --releases-dir .\
+                 component --component-name ${component["name"]}\
+                 get --version ${component["release"]["version"]["version"]} --pred || true
+            """,
+            returnStdout: true
+          )
+
+          println "=== predecessor component CLI standard out ==="
+          println pred_component_text
+
+          pred_component = readYaml text: pred_component_text
+        }
+
+        ORG="rcbops"
```

Comment:
Do we need to temporarily flip this to my username until we start merging all this into rcbops?

Reply:
No, it doesn't do anything. If you follow it through it's only used to print a string. I wasn't sure what would happen if RE-Release was used to tag on your repo, given we know it works for rcbops repos I've just got it printing the command it would use when we're ready to release.